app/Helpers/database/site-award.php

```diff
@@ -211,80 +195,84 @@ function SetCertifiedLegend(User $user, bool $enable): void
  */
 function getRecentProgressionAwardData(
     string $date,
-    ?string $friendsOf = null,
+    ?User $friendsOfUser = null,
     int $offset = 0,
     int $count = 50,
     ?int $onlyAwardType = null,
     ?int $onlyUnlockMode = null,
 ): array {
-    // Determine the friends condition
-    $friendCondAward = "";
-    if ($friendsOf !== null) {
-        $friendSubquery = GetFriendsSubquery($friendsOf);
-        $friendCondAward = "AND saw.User IN ($friendSubquery)";
-    }
+    $startOfDay = Carbon::createFromFormat("Y-m-d", $date)->startOfDay();
+    $endOfDay = Carbon::createFromFormat("Y-m-d", $date)->endOfDay();
+
+    $initialResults = PlayerBadge::with(['user', 'gameIfApplicable.system'])
+        ->whereBetween('AwardDate', [$startOfDay, $endOfDay])
+        ->when($onlyAwardType !== null, fn ($query) => $query->where('AwardType', $onlyAwardType))
+        ->when($onlyUnlockMode !== null, fn ($query) => $query->where('AwardDataExtra', $onlyUnlockMode))
+        ->when($friendsOfUser !== null, fn ($query) => $query->whereIn('user_id', $friendsOfUser->followedUsers()->pluck('UserAccounts.id')->toArray())
+        )
+        ->orderBy('AwardDate', 'DESC')
+        ->offset($offset)
+
+        // We don't want to show multiple rows per user for the same game if they earned
+        // multiple awards very quickly (ie: beat a game and also mastered it simultaneously).
+        // Unfortunately, using window functions like ROW_NUMBER() with Eloquent ORM is
+        // a huge hassle and arguably makes this code much more difficult to maintain. Rather than
+        // try to use a window function, we'll slightly overfetch from the database, then manually
+        // filter the records down later.
```

Comment:
Want to call out this change specifically - I spent about an hour fiddling around with getting `ROW_NUMBER()` to work with this Eloquent ORM migration. Ultimately, I decided the complexity trade-off really wasn't worth it, so I fell back to this overfetching solution.

Reply:
Does this work correctly with paging? If It takes 60 records to display 50 on the first page, will going to page 2 offset by 60 or 50?

Reply:
It doesn't - I made an oversight.

Getting this function to work with Eloquent ORM has proven to be quite challenging. I've reverted it back to the SQL implementation and adjusted the read to be from `user_id` instead of `User`. We can handle the Eloquent ORM migration later as a separate piece of work.